### ISSUE TITLE
[BUG] - Calling `StartAsync()` - `Stop()` - `StartAsync()` sporadically panics: `panic: close of closed channel`

### ISSUE THREAD

**Issue by mertyildiran** [go-co-op/gocron#425]
### Describe the bug

Calling `StartAsync()` - `Stop()` - `StartAsync()` in this order, causes sporadic panics: `panic: close of closed channel`

### To Reproduce

Minimal example:

```go
package main

import (
	"fmt"
	"time"

	"github.com/go-co-op/gocron"
)

func main() {
	s := gocron.NewScheduler(time.UTC)
	s.Every(5).Seconds().Do(func() {
		fmt.Println("Hello world")
	})
	s.StartAsync()
	s.Stop()
	s.StartAsync()
}
```

`go.mod` file:

```go
module main

go 1.17

require github.com/go-co-op/gocron v1.18.0

require (
	github.com/robfig/cron/v3 v3.0.1 // indirect
	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c // indirect
)
```

Build and run:

```shell
go build -o main main.go && repeat 10 { ./main }
```

Output:

```
Hello world
Hello world
panic: close of closed channel

goroutine 7 [running]:
github.com/go-co-op/gocron.(*executor).start(0xc0000a0300)
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/executor.go:118 +0x23f
created by github.com/go-co-op/gocron.(*Scheduler).start
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/scheduler.go:89 +0x6f
panic: close of closed channel

goroutine 7 [running]:
github.com/go-co-op/gocron.(*executor).start(0xc0000a0300)
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/executor.go:118 +0x23f
created by github.com/go-co-op/gocron.(*Scheduler).start
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/scheduler.go:89 +0x6f
Hello world
Hello world
panic: close of closed channel

goroutine 19 [running]:
github.com/go-co-op/gocron.(*executor).start(0xc000120300)
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/executor.go:118 +0x23f
created by github.com/go-co-op/gocron.(*Scheduler).start
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/scheduler.go:89 +0x6f
Hello world
Hello world
panic: close of closed channel

goroutine 7 [running]:
github.com/go-co-op/gocron.(*executor).start(0xc0000a0300)
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/executor.go:118 +0x23f
created by github.com/go-co-op/gocron.(*Scheduler).start
	/home/mertyildiran/go/pkg/mod/github.com/go-co-op/gocron@v1.18.0/scheduler.go:89 +0x6f
```

Panic happens on [this line](https://github.com/go-co-op/gocron/blob/9aaee3a5cd42aa105bb396ff829326f479cda2e4/executor.go#L118).

### Version

`v1.18.0`: https://github.com/go-co-op/gocron/releases/tag/v1.18.0

### Expected behavior

Shouldn't panic and I should be able re-start the scheduler with `StartAsync()` after I call `Stop()`.

**Comment by JohnRoesler** [comment]
Hm, yes it shouldn't do that. However, I wonder if calling them back to back like the example is too fast that it is causing the issue. Does it happen in a normal use case where you start and stop the scheduler at different times?

**Comment by birkoffe** [comment]
same bug with `fsnotify` which update cron jobs on file changing

```
func update() {
    s.Stop()
    s.Clear()

    // load config from file

    for _, job := range jobs {
        // 
    }

    s.StartAsync()
}
```

**Comment by mertyildiran** [comment]
> Hm, yes it shouldn't do that. However, I wonder if calling them back to back like the example is too fast that it is causing the issue. Does it happen in a normal use case where you start and stop the scheduler at different times?

I have separate HTTP endpoints calling `StartAsync` and `Stop` and they too crash sporadically.

**Comment by mertyildiran** [comment]
@JohnRoesler I have added the `go.mod` file and output into the issue description which I forgot to include.

**Comment by Shehats** [comment]
Chiming in here, hi 👋 , I was building using this package and I noticed this, and it was concerning given what I was working on. Ended not going the async approach in the end.

There is a race condition happening there, because the stop function is utilizing the `s.startBlockingStopChanMutex` which is used in the case of `StartBlocking`. I have raised a [PR](https://github.com/go-co-op/gocron/pull/434) for this :).
